### ISSUE TITLE
Zig Test exits with status code 255 (-1) when running under gdb/lldb

### ISSUE THREAD

**Issue by iansimonson** [ziglang/zig#5550]
Given the following file called test.zig:
```
test "a test" {
    var x: i32 = 3;
}
```

`zig test test.zig` results in:
```
All 1 tests passed.
```

However when running under lldb/gdb the result is:
```
(lldb) r test test.zig
Process 12429 launched: '/usr/local/bin/zig' (x86_64)
All 1 tests passed.

Tests failed. Use the following command to reproduce the failure:
/path/to/the/zig-cache/o/k2nUiPUWfTOWnk_unKQCys_HgfsDT2gOANLjEeifMgPe80Y_4x1-qZ_4LWnZU7jM/test
Process 12429 exited with status = 255 (0x000000ff)
```

This seems to be due to the child process exiting with `TerminationIdSignaled` rather than `TerminationIdClean` in main.cpp:1830

**Comment by Vexu** [comment]
Cannot reproduce.
```
(gdb) r test a.zig
Starting program: /usr/bin/zig test a.zig
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[Detaching after vfork from child process 624571]
All 1 tests passed.
[Inferior 1 (process 624558) exited normally]
```
```
(lldb) target create "zig"
Current executable set to 'zig' (x86_64).
(lldb) r test a.zig
Process 624713 launched: '/usr/bin/zig' (x86_64)
All 1 tests passed.
Process 624713 stopped and restarted: thread 1 received signal: SIGCHLD
Process 624713 exited with status = 0 (0x00000000) 
```

**Comment by iansimonson** [comment]
Additional info: Running on a Macbook Pro, macOS Catalina version 10.15.4

```
iansimonson@A-Macbook ~/D/z/single_files [1]> lldb
(lldb) target create zig
Current executable set to 'zig' (x86_64).
(lldb) r test test.zig
Process 83903 launched: '/usr/local/bin/zig' (x86_64)
All 1 tests passed.

Tests failed. Use the following command to reproduce the failure:
/Users/iansimonson/Development/zig/single_files/zig-cache/o/k2nUiPUWfTOWnk_unKQCys_HgfsDT2gOANLjEeifMgPe80Y_4x1-qZ_4LWnZU7jM/test
Process 83903 exited with status = 255 (0x000000ff)
(lldb)
```

backtrace when setting a breakpoint at the exit function:

```
(lldb) breakpoint set --name main_exit --condition 'exit_code == -1'
Breakpoint 1: where = zig`main_exit(Stage2ProgressNode*, int) + 15 at main.cpp:255:28, address = 0x000000010000255f
(lldb) r test test.zig
Process 83985 launched: '/usr/local/bin/zig' (x86_64)
All 1 tests passed.

Tests failed. Use the following command to reproduce the failure:
/Users/iansimonson/Development/zig/single_files/zig-cache/o/k2nUiPUWfTOWnk_unKQCys_HgfsDT2gOANLjEeifMgPe80Y_4x1-qZ_4LWnZU7jM/test
Process 83985 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 1.1
    frame #0: 0x000000010000255f zig`main_exit(root_progress_node=0x0000000000000000, exit_code=-1) at main.cpp:255:28
   252  #endif
   253
   254  int main_exit(Stage2ProgressNode *root_progress_node, int exit_code) {
-> 255      if (root_progress_node != nullptr) {
   256          stage2_progress_end(root_progress_node);
   257      }
   258      return exit_code;
Target 0: (zig) stopped.
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 1.1
  * frame #0: 0x000000010000255f zig`main_exit(root_progress_node=0x0000000000000000, exit_code=-1) at main.cpp:255:28
    frame #1: 0x0000000100009beb zig`main0(argc=3, argv=0x00007ffeefbff918) at main.cpp:1830:24
    frame #2: 0x00000001000025b4 zig`main(argc=3, argv=0x00007ffeefbff918) at main.cpp:1858:19
    frame #3: 0x00007fff73b7ecc9 libdyld.dylib`start + 1
    frame #4: 0x00007fff73b7ecc9 libdyld.dylib`start + 1
(lldb) p term
(Termination) $1 = (how = TerminationIdSignaled, code = 126)
```

**Comment by andrewrk** [comment]
This issue can be closed once someone tries this on macos 10.15.4 or later under and it works as expected.

**Comment by iansimonson** [comment]
Can confirm this works as expected for at least 10.15.6
